complaints/ccdb/index_ccdb.py

```diff
@@ -1,13 +1,79 @@
 import configargparse
 import os
 import sys
+import time
 import json
 from common.es_proxy import add_basic_es_arguments, get_es_connection
 from common.log import setup_logging
+from datetime import datetime
 from elasticsearch import TransportError
 from elasticsearch.helpers import bulk
 
 
+# -----------------------------------------------------------------------------
+# Enhancing Functions
+# -----------------------------------------------------------------------------
+
+def parse_date(date_str):
+    if not date_str:
+        return None
+
+    for fmt in ['%Y-%m-%dT%H:%M:%S', '%Y-%m-%d']:
+        try:
+            return datetime.strptime(date_str, fmt)
+        except ValueError:
+            pass
+
+    return None
+
+
+def format_date_est(date_str):
+    """format the date at noon Eastern Standard Time"""
+    d = parse_date(date_str)
+    if not d:
+        return None
+    return d.strftime("%Y-%m-%d") + 'T12:00:00-05:00'
+
+
+def format_date_as_mdy(date_str):
+    d = parse_date(date_str)
+    if not d:
+        return None
+    return d.strftime("%m/%d/%y")
+
+
+def enhance_complaint(complaint):
+    if 'complaint_id' not in complaint:
+        complaint['complaint_id'] = complaint['public_id']
+
+    # Already provided by streamParser.py
+    if ':updated_at' in complaint:
+        return complaint
+
+    # Simulate the Socrata field
+    dt = parse_date(complaint.get("date_received"))
+    complaint[":updated_at"] = time.mktime(dt.timetuple()) if dt else 0.0
```

Comment:
Because the synthetic ":updated_at" field is the last time the record was updated in the Socrata system, I would recommend using the date / time of the pipeline run to update this value, rather than the date_received of the complaint, but I will dig into this further.

More docs on the value we're attempting to simulate: https://dev.socrata.com/docs/system-fields.html